src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversion.cs

```diff
@@ -67,6 +67,18 @@ internal bool IsArrayIndex
             }
         }
 
+        private class DeconstructionUncommonData : UncommonData
```

Comment:
If you are not using anything from UncommonData, perhaps making this a sibling of UncommonData derived from a common base would be more appropriate? - just to save on unnecessary fields.